### ISSUE TITLE
Problem about the loss function

### ISSUE THREAD

**Issue by Dawn-LX** [alirezazareian/vspnet#9]
I am confused about the loss function of entity and predicate.
In your paper, Eq.(10) and Eq.(11), you use the l2_loss between E^O and E^T

**However**, in the code, it seems like that you calculate the similarity (dot product) between the entity embedding and the ground-truth entity category word embedding. i.e., 
https://github.com/alirezazareian/vspnet/blob/68db0265b49107e0aa66430eea4cf56faa85aece/src/tensorflow/models/vsp_model_002.py#L391

and then you use cross-entropy loss at 
https://github.com/alirezazareian/vspnet/blob/68db0265b49107e0aa66430eea4cf56faa85aece/src/tensorflow/models/vsp_model_002.py#L438

This seems to be different from that is described in the paper (Eq.10 & Eq.11). Why?
looking forward to you answer! thank you!

**Comment by alirezazareian** [comment]
In the paragraph after Eq. (9), we mentioned L2 as just an example of the loss that can be used, since the alignment and weakly supervised training framework is independent of the type of loss you define for each node/edge. I have tried both L2 and cross-entropy, and both perform similarly well. L2 is particularly useful for zero-shot settings, which was out of the scope of the paper, and we did not include experiments with that. Cross-entropy is more straightforward in the current setting since it is a closed-set classification problem.

**Comment by Dawn-LX** [comment]
> t include exper

got it, thank you~

**Comment by Dawn-LX** [comment]
> In the paragraph after Eq. (9), we mentioned L2 as just an example of the loss that can be used, since the alignment and weakly supervised training framework is independent of the type of loss you define for each node/edge. I have tried both L2 and cross-entropy, and both perform similarly well. L2 is particularly useful for zero-shot settings, which was out of the scope of the paper, and we did not include experiments with that. Cross-entropy is more straightforward in the current setting since it is a closed-set classification problem.

got it, thank you~
